### PR TITLE
fix(installer): fix various installer issues

### LINKED ISSUE #11553: Profile field migration update created on clean 3.0 install

### LINKED ISSUE #11935: Elgg 3.0 installation issues
Fresh install using the website (no `settings.php` present)
- after entering the database settings (username, pwd, etc) the next step fails with an error 'couldn't connect to database' hitting refresh on the form will continue the installation as expected
- no plugins active

Fresh install using the website (`settings.php` present)
- no plugins active

Fresh install using `elgg-cli` (no `settings.php` present)
- no installation issues
- upgrades not registered (go to upgrades page will register them)
- activate new plugins upgrade is available
- deactivate old plugins upgrade is available
- migrate profile fields upgrade is available
- running `upgrade.php` cleared new and old plugins upgrade, profile fields remain. Running `elgg-cli upgrade` directly after `elgg-cli install` has the same effect

### PR DESCRIPTION
**Pull request by hypeJunction**
Caches are no longer enabled during installation.
Database connection is correctly established on first try.
Plugins are activated on install as they should.

Fixes #11935
Fixes #11553

### CODE REVIEW

**Comment by hypeJunction** [comment]
@jeabakker @jdalsem If you are not going to work on all the wanna-haves in the milestone, please remove them. Let's see where we are. I want RC out asap, otherwise we keep dragging it on, and adding more bugs along with new features.

File: engine/classes/ElggInstaller.php [hunk 1]
```diff
@@ -113,6 +114,11 @@ protected function getApp() {
 			$config->elgg_config_locks = false;
 			$config->installer_running = true;
 			$config->dbencoding = 'utf8mb4';
+			$config->boot_cache_ttl = 0;
+			$config->system_cache_enabled = false;
+			$config->simplecache_enabled = false;
+			$config->debug = \Psr\Log\LogLevel::WARNING;
+			$config->cacheroot = Paths::project() . 'install/cache/';
```
Comment (jeabakker) [inline engine/classes/ElggInstaller.php]: What if project path isn't writeable (as it should be)? Wouldn't `sys_get_temp_dir()` be better?

Comment (hypeJunction) [reply engine/classes/ElggInstaller.php]: I am not sure what `sys_get_temp_dir` is supposed to do. Last time I tried to use TempFile class you added, it didn't work for me.

Comment (jeabakker) [reply engine/classes/ElggInstaller.php]: https://secure.php.net/manual/en/function.sys-get-temp-dir.php

Comment (jdalsem) [reply engine/classes/ElggInstaller.php]: @hypeJunction i think the problem @jeabakker describes is valid. Can you agree with the suggestion?

Comment (hypeJunction) [reply engine/classes/ElggInstaller.php]: I am not going to spend any more time on this. Without setting cacheroot it just writes the cache into root anyway.

Comment (hypeJunction) [reply engine/classes/ElggInstaller.php]: > Can you agree with the suggestion?
As I said, last time I tried to use `TempFile` that uses this approach it didn't work, so I am not sure if it's a valid approach. But feel free to fork and do as you please.

Comment (jdalsem) [reply engine/classes/ElggInstaller.php]: project root is not writeable as we suggest for security reasons, so this PR will cause issues (as does the current implementation).

Comment (jdalsem) [reply engine/classes/ElggInstaller.php]: > last time I tried to use TempFile that uses this approach it didn't work

i am unaware of a problem with this class... could you create a ticket for it?

Comment (hypeJunction) [reply engine/classes/ElggInstaller.php]: Whatever

**Pull request closed without merging**